### PR TITLE
Update PHP linting rules

### PR DESCRIPTION
**Pull request by jrtashjian**
Resolved all PHPCS warnings and errors. Requires review.

### CODE REVIEW

**Comment by AnthonyLedesma** [comment]
Going to start working this.

File: phpcs.xml [hunk 1]
```diff
@@ -1,15 +1,77 @@
...
+		-->
+		<exclude name="WordPress.Files.FileName.InvalidClassFileName"/>
+		<exclude name="WordPress.Files.FileName.NotHyphenatedLowercase"/>
+		<!--
+		ShortPrefixPassed is excluded due to the use of prefix `Go` This prefix, 
+		while short in length, is approved by the themes team at WordPress.org.
+		For this reason the rule should be excluded as we are not concerned with 
+		code collisions.
+		-->
+		<exclude name="WordPress.NamingConventions.PrefixAllGlobals.ShortPrefixPassed" />
```
Comment (AnthonyLedesma) [inline phpcs.xml:38]: Added this snipped to ignore the short prefix error thrown by PHPCS. The other route we can go with this is to increase the length of the prefix. Maybe something like `gdgo`?

File: includes/amp.php [hunk 2]
```diff
@@ -52,12 +52,11 @@ function amp_body_class( $classes ) {
  *
  * @param string   $item_output The menu item's starting HTML output.
  * @param WP_Post  $item        Menu item data object.
- * @param int      $depth       Depth of menu item. Used for padding.
  * @param stdClass $args        An object of wp_nav_menu() arguments.
  *
  * @return string Modified nav menu item HTML.
  */
-function amp_nav_sub_menu_buttons( $item_output, $item, $depth, $args ) {
```
Comment (AnthonyLedesma) [inline includes/amp.php:60]: Should this be removed? We are not using the reference within the function. If we look at https://github.com/godaddy-wordpress/go/blob/ee70c2dfcafe7dfddc727c8f3c52e94898db966c/includes/core.php#L1213, for example, we can see the value properly used within a function.

Comment (EvanHerman) [reply includes/amp.php:60]: I would just ignore that line so it passes in the tests. I think we still need it present.

File: phpcs.xml (hunk 1 above)
Comment (EvanHerman) [reply phpcs.xml:38]: I ignored those in all of the Primer themes as well. It doesn't seem worth it to go through and append a prefix onto every method. We're also namespacing our files so there shouldn't be conflicts with plugins.

**Pull request merged**